### ISSUE TITLE
Implement public API for manipulating table matching 

### ISSUE THREAD

**Issue by ArnCarveris** [SanderMertens/flecs#109]
**Describe the problem you are trying to solve.**
I'm always frustrated that in `flecs` there is no efficient and clear API for processing only specific subset of data with specific way when game content is already loaded. 

Because of that, there is not that clean and easy to implement such systems like: 
 * `state-machine`: edit/play game states switching in run-time like in [CryEngine Sandbox Editor](https://www.cryengine.com/features/view/sandbox)
 * `multi-level`: when all needed in near future levels are loaded but you processing only one level
 * `world-streaming`: when asynchronously loading/unloading game world while travelling, read [ OWMAN](https://github.com/ArnCarveris/OWMAN/blob/master/doc/be.pdf)
 * `occlusion-culling`: like in [Unity sectr-vis plugin](http://www.procedural-worlds.com/sectr/sectr-vis/) and like in [CryEngine VisArea+Portals](https://docs.cryengine.com/display/SDKDOC2/VisArea+and+Portals)

**Describe the solution you'd like**
To summarize there is following requirements for API:
* It should be possible to quickly turn on/off matched tables per system
* It should be possible to quickly turn on/off matched tables per world
* Turning on/off should not be an O(n) operation (where N is number of tables)
* Table selection could either be tag based or type filter based

**Comment by SanderMertens** [comment]
This capability will be realized by #295. These new queries will allow for parameterized terms, which will allow for quick switching between different datasets. The feature will satisfy the requirements in the described solution. I will close this issue, let's continue discussion for this issue on #295
